applications/MappingApplication/custom_utilities/interface_condition.h

```diff
@@ -93,89 +89,206 @@ namespace Kratos
+                  && mNumPoints == 2) { // I am a linear line condition
+                  is_inside = MapperUtilities::ProjectPointToLine(mpCondition, rGlobalCoords,
                                                                   projection_local_coords,
-                                                                  projection_distance, m_normal);
-          } else if (m_geometry_family == GeometryData::Kratos_Quadrilateral) { // I am a quadrilateral condition
-              is_inside = MapperUtilities::ProjectPointToQuaddrilateral(m_p_condition, global_coords,
-                                                                        projection_local_coords,
-                                                                        projection_distance, m_normal);
-          } else {
-              KRATOS_ERROR << "MappingApplication; InterfaceCondition; \"EvaluateResult\" Unsupported Geometry!" << std::endl;
+                                                                  projection_distance);
+              } else if (mGeometryFamily == GeometryData::Kratos_Triangle 
+                        && mNumPoints == 3) { // I am a linear triangular condition
+                  is_inside = MapperUtilities::ProjectPointToTriangle(mpCondition, rGlobalCoords,
+                                                                      projection_local_coords,
+                                                                      projection_distance);
+              } else if (mGeometryFamily == GeometryData::Kratos_Quadrilateral
+                        && mNumPoints == 4) { // I am a linear quadrilateral condition
+                  is_inside = MapperUtilities::ProjectPointToQuadrilateral(mpCondition, rGlobalCoords,
+                                                                          projection_local_coords,
+                                                                          projection_distance);
+              } else if (mGeometryFamily == GeometryData::Kratos_Tetrahedra ||
+                         mGeometryFamily == GeometryData::Kratos_Prism ||
+                         mGeometryFamily == GeometryData::Kratos_Hexahedra) { // Volume Mapping
+                  is_inside = MapperUtilities::PointLocalCoordinatesInVolume(mpCondition, rGlobalCoords,
+                                                                             projection_local_coords,
+                                                                             projection_distance);
+              } else {
+                  std::cout << "MAPPER WARNING, Unsupported geometry, " 
+                            << "using an approximation" << std::endl;
+                  return false;
+              }
+          } catch (...) {
+              std::cout << "MAPPER WARNING, Functions necessary for projection "
+                        << "are not implemented in the core, " 
+                        << "using an approximation" << std::endl;
+              return false;
           }
```

Comment:
I understand it now. Still, I would prefer not to rely on the try/catch thing, since it can hide other issues not related to what you mention for the geometry. My preferred solution here would be to do a more strict filtering of the geometry type, so that we cannot call non-implemented functions.
I would propose to try to filter the geometries based on the 'GetGeometryType`method instead of `GetGeometryFamily`(the possible return types are listed on the `KratosGeometryType`enum of geometry_data.h) or, alternatively, the GeometryFamily in combination with the number of points, so that you only call the methods for the geometries that are supported and return false otherwise. Do you think this could be a workable solution?